library/std/src/sys/common/alloc.rs

```diff
@@ -28,6 +28,13 @@ pub const MIN_ALIGN: usize = 8;
     target_arch = "wasm64",
 )))]
 pub const MIN_ALIGN: usize = 16;
+// By default, the allocator on the esp-idf platform guarentees no specific alignment.
+// This can be changed, but not though libc API's so we set the MIN_ALIGN here to 1.
```

Comment:
There's a minor typo in "though", it should be "through" also i think "API's" should be "APIs"